includes/class-wc-ajax.php

```diff
@@ -1306,6 +1306,13 @@ public static function product_ordering() {
 			$index ++;
 			$menu_orders[ $id ] = $index;
 			$wpdb->update( $wpdb->posts, array( 'menu_order' => $index ), array( 'ID' => $id ) );
+			
+			/**
+			 * When a single product has gotten it's ordering updated.
+			 * $id The product ID
+			 * $index The new menu order
+			*/
+			do_action( 'woocommerce_after_single_product_ordering', $id, $index );
```

Comment:
Thats fine for your use case, but it doesn't fit other potential uses? Therefore (as is) it doesn't seem fit for core.